lib/rspec/core/bisect/runner.rb

```diff
@@ -71,24 +72,45 @@ def run_locations(*capture_args)
         # https://github.com/jruby/jruby/issues/2766
         if Open3.respond_to?(:capture2e) && !RSpec::Support::Ruby.jruby?
           def run_command(cmd)
-            Open3.capture2e(cmd).first
+            Open3.capture2e(bisect_environment_hash, cmd).first
           end
         else # for 1.8.7
           # :nocov:
           def run_command(cmd)
             out = err = nil
 
+            original_spec_opts = ENV['SPEC_OPTS']
+            ENV['SPEC_OPTS'] = spec_opts_without_bisect
+
```

Comment:
The upside of supporting ruby 1.8.7 is that I get to use the old hash syntax without feeling guilt. The downside is that some of stdlib wasn't as capable back then, and `Open3`/`Process.spawn` is [one of those bits](http://ruby-doc.org/stdlib-1.8.7/libdoc/open3/rdoc/Open3.html), unfortunately. I've used the new passed environment [where possible](https://github.com/urbanautomaton/rspec-core/blob/21ed282ce87103c9b5db4fd06056d13ffde997c8/lib/rspec/core/bisect/runner.rb#L73-L76), but for 1.8.7 it's either put the new env in the command string or mutate `ENV`, so I plumped for the latter.